src/trace-api.js

```diff
@@ -197,6 +199,23 @@ TraceAgent.prototype.getCurrentContextId = function() {
 };
 
 /**
+ * Returns the projectId that was either configured or auto-discovered by the
+ * TraceWriter. Note that the auto-discovery is done asynchronously, so this
+ * will return null until the projectId auto-discovery completes.
+ */
+TraceAgent.prototype.getWriterProjectId = function() {
+  var traceWriter;
+  try {
+     traceWriter = TraceWriter.get();
```

Comment:
The current behavior is that `TraceWriter.get` will fail if the writer has not been created with a call to `TraceWriter.create`. However, even after create is called, the project id may not be available until it has been resolved here: https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/blob/master/src/trace-writer.js#L99. I would recommend returning the value of `this.config_.projectId`. If this value is truthy, it will be used as the project id for the lifetime of the trace agent and the metadata service will never be consulted. If it is falsey, it will be overwritten with the asynchronously resolved id from the metadata service.

Reply:
Done. That feels much cleaner.